### PR TITLE
Rocky: Add ALLOWED_HOSTS, CSRF_TRUSTED_ORIGINS env vars

### LINKED ISSUE #1025: Django upgrade to 4.2 breaks CSRF
**Describe the bug**
Just tried to upgrade to the latest version with https://github.com/minvws/nl-kat-coordination/commit/6a2ca64d5c3d6f8fe84657accba48d931db81677 and CSRF seems broken.

**To Reproduce**
1. pull main
2. make kat
3. login

**Expected behavior**
I expect to be able to login and the octopoes_api to work.

**logs**

***rocky logs***
```
rocky_1                | [23/May/2023 12:44:12] "GET / HTTP/1.0" 302 0
rocky_1                | [23/May/2023 12:44:12] "GET /nl/ HTTP/1.0" 200 5087
rocky_1                | [23/May/2023 12:44:12] "GET /static/img/kat_logo.png HTTP/1.0" 200 46258
rocky_1                | [23/May/2023 12:44:12] "GET /static/js/dropdown.js HTTP/1.0" 200 818
rocky_1                | WARNING:opentelemetry.exporter.otlp.proto.grpc.exporter:Transient error StatusCode.UNAVAILABLE encountered while exporting traces, retrying in 8s.
rocky_1                | [23/May/2023 12:44:14] "GET /account/login/ HTTP/1.0" 302 0
rocky_1                | [23/May/2023 12:44:14] "GET /nl/login/ HTTP/1.0" 200 6027
rocky_1                | Forbidden (Origin checking failed - https://public-ip does not match any trusted origins.): /nl/login/
rocky_1                | WARNING:django.security.csrf:Forbidden (Origin checking failed - https://public-ip does not match any trusted origins.): /nl/login/
rocky_1                | [23/May/2023 12:44:15] "POST /nl/login/ HTTP/1.0" 403 2572
rocky_1                | WARNING:opentelemetry.exporter.otlp.proto.grpc.exporter:Transient error StatusCode.UNAVAILABLE encountered while exporting traces, retrying in 16s.
```

***octpoes_api logs***
```
octopoes_api_1         | [2023-05-23 12:39:00 +0000] [7] [ERROR] [io_services_utils] Socket failed to connect: <socket.socket fd=18, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=6, laddr=('172.22.0.12', 35046)>; error=111 (Connection refused)
octopoes_api_1         | [2023-05-23 12:39:00 +0000] [7] [ERROR] [connection_workflow] TCP Connection attempt failed: ConnectionRefusedError(111, 'Connection refused'); dest=(<AddressFamily.AF_INET: 2>, <SocketKind.SOCK_STREAM: 1>, 6, '', ('172.22.0.3', 5672))
octopoes_api_1         | [2023-05-23 12:39:00 +0000] [7] [ERROR] [connection_workflow] AMQPConnector - reporting failure: AMQPConnectorSocketConnectError: ConnectionRefusedError(111, 'Connection refused')
octopoes_api_1         | [2023-05-23 12:39:00 +0000] [7] [ERROR] [connection_workflow] AMQP connection workflow failed: AMQPConnectionWorkflowFailed: 1 exceptions in all; last exception - AMQPConnectorSocketConnectError: ConnectionRefusedError(111, 'Connection refused'); first exception - None.
octopoes_api_1         | [2023-05-23 12:39:00 +0000] [7] [ERROR] [connection_workflow] AMQPConnectionWorkflow - reporting failure: AMQPConnectionWorkflowFailed: 1 exceptions in all; last exception - AMQPConnectorSocketConnectError: ConnectionRefusedError(111, 'Connection refused'); first exception - None
octopoes_api_1         | [2023-05-23 12:39:00 +0000] [7] [ERROR] [blocking_connection] Connection workflow failed: AMQPConnectionWorkflowFailed: 1 exceptions in all; last exception - AMQPConnectorSocketConnectError: ConnectionRefusedError(111, 'Connection refused'); first exception - None
octopoes_api_1         | [2023-05-23 12:39:00 +0000] [7] [ERROR] [blocking_connection] Error in _create_connection().
octopoes_api_1         | Traceback (most recent call last):
octopoes_api_1         |   File "/usr/local/lib/python3.8/site-packages/pika/adapters/blocking_connection.py", line 451, in _create_connection
octopoes_api_1         |     raise self._reap_last_connection_workflow_error(error)
octopoes_api_1         | pika.exceptions.AMQPConnectionError
octopoes_api_1         | [2023-05-23 12:39:00 +0000] [7] [CRITICAL] [api] 
octopoes_api_1         | [2023-05-23 12:39:00 +0000] [7] [INFO] [h11_impl] 172.22.0.13:53002 - "GET /demo/objects?types=DNSZone&scan_level=1&scan_level=2&scan_level=3&scan_level=4&offset=0&limit=1 HTTP/1.1" 500
octopoes_api_1         | [2023-05-23 12:39:00 +0000] [7] [ERROR] [h11_impl] Exception in ASGI application
octopoes_api_1         | Traceback (most recent call last):
octopoes_api_1         |   File "/usr/local/lib/python3.8/site-packages/uvicorn/protocols/http/h11_impl.py", line 407, in run_asgi
octopoes_api_1         |     result = await app(  # type: ignore[func-returns-value]
octopoes_api_1         |   File "/usr/local/lib/python3.8/site-packages/uvicorn/middleware/proxy_headers.py", line 78, in __call__
octopoes_api_1         |     return await self.app(scope, receive, send)
octopoes_api_1         |   File "/usr/local/lib/python3.8/site-packages/uvicorn/middleware/message_logger.py", line 86, in __call__
octopoes_api_1         |     raise exc from None
octopoes_api_1         |   File "/usr/local/lib/python3.8/site-packages/uvicorn/middleware/message_logger.py", line 82, in __call__
octopoes_api_1         |     await self.app(scope, inner_receive, inner_send)
octopoes_api_1         |   File "/usr/local/lib/python3.8/site-packages/fastapi/applications.py", line 270, in __call__
octopoes_api_1         |     await super().__call__(scope, receive, send)
octopoes_api_1         |   File "/usr/local/lib/python3.8/site-packages/starlette/applications.py", line 124, in __call__
octopoes_api_1         |     await self.middleware_stack(scope, receive, send)
octopoes_api_1         |   File "/usr/local/lib/python3.8/site-packages/starlette/middleware/errors.py", line 184, in __call__
octopoes_api_1         |     raise exc
octopoes_api_1         |   File "/usr/local/lib/python3.8/site-packages/starlette/middleware/errors.py", line 162, in __call__
octopoes_api_1         |     await self.app(scope, receive, _send)
octopoes_api_1         |   File "/usr/local/lib/python3.8/site-packages/opentelemetry/instrumentation/asgi/__init__.py", line 579, in __call__
octopoes_api_1         |     await self.app(scope, otel_receive, otel_send)
octopoes_api_1         |   File "/usr/local/lib/python3.8/site-packages/starlette/middleware/base.py", line 106, in __call__
octopoes_api_1         |     response = await self.dispatch_func(request, call_next)
octopoes_api_1         |   File "/usr/local/lib/python3.8/site-packages/fastapi_utils/timing.py", line 47, in timing_middleware
octopoes_api_1         |     response = await call_next(request)
octopoes_api_1         |   File "/usr/local/lib/python3.8/site-packages/starlette/middleware/base.py", line 80, in call_next
octopoes_api_1         |     raise app_exc
octopoes_api_1         |   File "/usr/local/lib/python3.8/site-packages/starlette/middleware/base.py", line 69, in coro
octopoes_api_1         |     await self.app(scope, receive_or_disconnect, send_no_error)
octopoes_api_1         |   File "/usr/local/lib/python3.8/site-packages/starlette/middleware/exceptions.py", line 79, in __call__
octopoes_api_1         |     raise exc
octopoes_api_1         |   File "/usr/local/lib/python3.8/site-packages/starlette/middleware/exceptions.py", line 68, in __call__
octopoes_api_1         |     await self.app(scope, receive, sender)
octopoes_api_1         |   File "/usr/local/lib/python3.8/site-packages/fastapi/middleware/asyncexitstack.py", line 21, in __call__
octopoes_api_1         |     raise e
octopoes_api_1         |   File "/usr/local/lib/python3.8/site-packages/fastapi/middleware/asyncexitstack.py", line 18, in __call__
octopoes_api_1         |     await self.app(scope, receive, send)
octopoes_api_1         |   File "/usr/local/lib/python3.8/site-packages/starlette/routing.py", line 706, in __call__
octopoes_api_1         |     await route.handle(scope, receive, send)
octopoes_api_1         |   File "/usr/local/lib/python3.8/site-packages/starlette/routing.py", line 276, in handle
octopoes_api_1         |     await self.app(scope, receive, send)
octopoes_api_1         |   File "/usr/local/lib/python3.8/site-packages/starlette/routing.py", line 66, in app
octopoes_api_1         |     response = await func(request)
octopoes_api_1         |   File "/usr/local/lib/python3.8/site-packages/fastapi/routing.py", line 225, in app
octopoes_api_1         |     solved_result = await solve_dependencies(
octopoes_api_1         |   File "/usr/local/lib/python3.8/site-packages/fastapi/dependencies/utils.py", line 529, in solve_dependencies
octopoes_api_1         |     solved = await solve_generator(
octopoes_api_1         |   File "/usr/local/lib/python3.8/site-packages/fastapi/dependencies/utils.py", line 454, in solve_generator
octopoes_api_1         |     return await stack.enter_async_context(cm)
octopoes_api_1         |   File "/usr/local/lib/python3.8/contextlib.py", line 568, in enter_async_context
octopoes_api_1         |     result = await _cm_type.__aenter__(cm)
octopoes_api_1         |   File "/usr/local/lib/python3.8/contextlib.py", line 171, in __aenter__
octopoes_api_1         |     return await self.gen.__anext__()
octopoes_api_1         |   File "/usr/local/lib/python3.8/site-packages/fastapi/concurrency.py", line 36, in contextmanager_in_threadpool
octopoes_api_1         |     raise e
octopoes_api_1         |   File "/usr/local/lib/python3.8/contextlib.py", line 131, in __exit__
octopoes_api_1         |     self.gen.throw(type, value, traceback)
octopoes_api_1         | pika.exceptions.AMQPConnectionErr
```

**OpenKAT version**
main

**Additional context**
[Related thread](https://stackoverflow.com/questions/70612439/csrf-failed-origin-checking-failed-http-localhost8000-does-not-match-any)

adding the `CSRF_TRUSTED_ORIGINS` fixes this problem.
```
ALLOWED_HOSTS = ["*"]
CSRF_TRUSTED_ORIGINS = ["http://127.0.0.1:8000","https://nginx-public-ip"]
```

### PR DESCRIPTION
**Pull request by praseodym**
### Changes
Add environment variables to configure Django ALLOWED_HOSTS and CSRF_TRUSTED_ORIGINS in Rocky.

### Issue link

Closes #1025.

### Proof
n/a

---

### Code Checklist
- [x] All the commits in this PR are properly PGP-signed and verified;
- [x] This PR only contains functionality relevant to the issue; tickets have been created for newly discovered issues.
- [ ] I have written unit tests for the changes or fixes I made.
- [ ] For any non-trivial functionality, I have added integration and/or end-to-end tests.
- [x] I have performed a self-review of my code and refactored it to the best of my abilities.

### Communication
- [x] I have informed others of any required `.env` changes files if required and changed the `.env-dist` accordingly.
- [x] I have made corresponding changes to the documentation, if necessary.

---
## Checklist for code reviewers:
Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_code.md) into your comment.

---
## Checklist for QA:
Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_qa.md) into your comment.

### CODE REVIEW

**Comment by praseodym** [comment]
Do we need a different default for `DJANGO_CSRF_TRUSTED_ORIGINS`, such as `http://localhost http://127.0.0.1`?

**Review by ammar92 — DISMISSED** [review]

**Comment by underdarknl** [comment]
Lets make a mental note that we also include these changes in our update docs for 1.10

**Comment by underdarknl** [comment]
Also, does gitpod still function out of the box with these settings?

File: rocky/rocky/settings.py [hunk 1]
```diff
@@ -256,6 +258,24 @@
...
+    # public url on https://8000-minvws-nlkatcoordinatio-fykdue22b07.ws-eu98.gitpod.io/
+    ALLOWED_HOSTS.append("8000-" + os.getenv("GITPOD_WORKSPACE_URL").split("//")[1])
+    CSRF_TRUSTED_ORIGINS.append(os.getenv("GITPOD_WORKSPACE_URL").replace("//", "//8000-"))
+
+# Configuration for GitHub Codespaces
+if os.getenv("GITHUB_CODESPACES_PORT_FORWARDING_DOMAIN"):
+    # example environment variable: GITHUB_CODESPACES_PORT_FORWARDING_DOMAIN=preview.app.github.dev
+    # public url on https://praseodym-organic-engine-9j6465vx3xgx6-8000.preview.app.github.dev/
+    ALLOWED_HOSTS.append("." + os.getenv("GITHUB_CODESPACES_PORT_FORWARDING_DOMAIN"))
+    CSRF_TRUSTED_ORIGINS.append("https://*." + os.getenv("GITHUB_CODESPACES_PORT_FORWARDING_DOMAIN"))
```
Comment (ammar92) [inline rocky/rocky/settings.py:277]: Can you please introduce variables for these two vars?
```python3
GITPOD_WORKSPACE_URL = os.getenv("GITPOD_WORKSPACE_URL")
if GITPOD_WORKSPACE_URL is not None:
    ....

....

GITHUB_CODESPACES_PORT_FORWARDING_DOMAIN = os.getenv("GITHUB_CODESPACES_PORT_FORWARDING_DOMAIN")
if GITHUB_CODESPACES_PORT_FORWARDING_DOMAIN is not None:
    ....

Comment (praseodym) [reply rocky/rocky/settings.py]: See the current version, I think that is cleaner?

Comment (ammar92) [reply rocky/rocky/settings.py]: Walrus ops 🚀

**Review by ammar92 — APPROVED** [review]

**Comment by Darwinkel** [comment]
## Checklist for QA:
- [x] I have checked out this branch, and successfully ran a fresh `make reset`.
- [x] I confirmed that there are no unintended functional regressions in this branch:
    - [x] I have managed to pass the onboarding flow
    - [x] Objects and Findings are created properly
    - [x] Tasks are created and completed properly
- [x] I confirmed that the PR's advertised `feature` or `hotfix` works as intended.

### What works:
* Github Codespaces works out of the box
* Gitpod works out of the box
* localhost works out of the box
* An unspecified hostname correctly causes an ALLOWED_HOSTS error
* A specified hostname doesn't cause an ALLOWED_HOSTS error

### What doesn't work:
* I receive 403 CSRF errors on `http://hi.openkat.local` even though it is specified in `DJANGO_CSRF_TRUSTED_ORIGINS`. Looks like this is because it's an insecure http request. I don't have a https reverse proxy ready to use on my machine.

**Review by underdarknl — APPROVED** [review]

**Pull request merged**